src/extrinsic/extra.rs

```diff
@@ -271,21 +272,28 @@ impl<T: System + Balances + Clone + Debug + Eq + Send + Sync> SignedExtra<T>
         tx_version: u32,
         nonce: T::Index,
         genesis_hash: T::Hash,
+        era_info: (Era, Option<T::Hash>),
     ) -> Self {
         DefaultExtra {
             spec_version,
             tx_version,
             nonce,
             genesis_hash,
+            era_info,
         }
     }
 
     fn extra(&self) -> Self::Extra {
+        let era_hash = if let Some(hash) = self.era_info.1 {
+            hash
+        } else {
+            self.genesis_hash
+        };
         (
             CheckSpecVersion(PhantomData, self.spec_version),
             CheckTxVersion(PhantomData, self.tx_version),
             CheckGenesis(PhantomData, self.genesis_hash),
-            CheckEra((Era::Immortal, PhantomData), self.genesis_hash),
+            CheckEra((self.era_info.0, PhantomData), era_hash),
```

Comment:
This is a bit clunky. Ideally we'd have `CheckEra(PhantomData, self.era_info)` but that seems tricky to make work?

Reply:
The inner tuple is what actually gets encoded, while the `era_hash` is not part of the `CheckEra` struct in Substrate and is just used here for implementing `additional_signed`. We could make a function like `create_check_era(PhantomData, self.era_info)`, but it seems like that would just create unnecessary code.

Substrate FRAME nodes do not need the hash because they have a map of canon block number => block hash, which we do not have